File: src/leetcode/2025/12/Solution345.cpp
#include <string>
#include <iostream>
using namespace std;

class Solution345
{
public:
    string reverseVowels(string s)
    {
        int left = 0;
        int right = s.length() - 1;
        string vowels = "aeiouAEIOU";

        while (left < right)
        {
            while (left < right && vowels.find(s[left]) == string::npos)
            {
                left++;
            }
            while (left < right && vowels.find(s[right]) == string::npos)
            {
                right--;
            }
            if (left < right)
            {
                swap(s[left], s[right]);
                left++;
                right--;
            }
        }
        return s;
    }
};

int main()
{
    Solution345 solution;

    string input0 = "IceCreAm";
    string output0 = solution.reverseVowels(input0);
    cout <<output0<<endl; // Output: "AmCrEeIc"

    string input1 = "hello";
    string output1 = solution.reverseVowels(input1);
    cout <<output1<<endl; // Output: "holle"

    string input2 = "leetcode";
    string output2 = solution.reverseVowels(input2);
    cout <<output2<<endl; // Output: "leotcede"



    return 0;
}

File: src/leetcode/2025/12/Solution345.java
public class Solution345 {
    public String reverseVowels(String s) {
        char[] chars = s.toCharArray();
        int left = 0;
        int right = chars.length - 1;
        String vowels = "aeiouAEIOU";

        while (left < right) {
            while (left < right && !vowels.contains(String.valueOf(chars[left]))) {
                left++;
            }
            while (left < right && !vowels.contains(String.valueOf(chars[right]))) {
                right--;
            }
            if (left < right) {
                char temp = chars[left];
                chars[left] = chars[right];
                chars[right] = temp;
                left++;
                right--;
            }
        }
        return new String(chars);
    }

    public static void main(String[] args) {
        Solution345 solution = new Solution345();
        System.out.println(solution.reverseVowels("IceCreAm")); // "AceCreIm"

        System.out.println(solution.reverseVowels("leetcode")); // "leotcede"
    }
}
